frontend/components/ArticleEditor/ArticleEditorItem/index.tsx

```diff
@@ -35,15 +36,23 @@ export default function ArticleEditorItem({
   imageName,
 }: EditorProps) {
   const [fileName, setFileName] = React.useState('');
+  const { showMessage } = useSnackbar();
+  const { t } = useTranslation(['common', 'news']);
 
   const [getPresignedPutUrlMutation] = useGetPresignedPutUrlMutation({
     variables: {
       fileName,
     },
+    onError: (error) => {
+      console.error(error.message);
+      if (error.message.includes('You do not have permission')) {
```

Comment:
There must be a better way to do this, maybe through custom error codes?

Reply:
Errors codes apparently exists by default: https://www.apollographql.com/docs/apollo-server/data/errors/

This looks cleaner imo:
```typescript
if (error.extensions.code === 'UNAUTHENTICATED') {
 ...
}
``` 

Might even be worth having helper functions:
```typescript
function unauthenticated(error: /*some type*/) {
  return error.extensions.code === 'UNAUTHENTICATED';
}
function badUserInput(error: /*some type*/) {
  return error.extensions.code === 'BAD_USER_INPUT';
}
...
if (unauthenticated(error)) {
 ...
}
```

Reply:
this is the solution I ended up going for
```typescript
const handleApolloError = (
  apolloError: ApolloError,
  showMessage: (
    message: string,
    severity: 'success' | 'info' | 'warning' | 'error'
  ) => void,
  t: TFunction,
  genericErrorKey: string = 'common:error',
) => {
  const { graphQLErrors, message } = apolloError;
  console.error(message);
  graphQLErrors.forEach((error) => {
    switch (error.extensions.code) {
      case 'FORBIDDEN':
        showMessage(t('common:no_permission_action'), 'error');
        break;
      default:
        showMessage(t(genericErrorKey), 'error');
        break;
    }
  });
};
```

Reply:
in hindsight perhaps this function should just return a translation key and the consumer can do as they wish with it...